streams/src/main/java/org/apache/kafka/streams/state/internals/ChangeLoggingSessionBytesStore.java

```diff
@@ -79,15 +94,21 @@ public void init(final StateStoreContext context, final StateStore root) {
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public void remove(final Windowed<Bytes> sessionKey) {
         wrapped().remove(sessionKey);
-        context.logChange(name(), SessionKeySchema.toBinary(sessionKey), null, context.timestamp());
+        context.logChange(name(), SessionKeySchema.toBinary(sessionKey), null, context.timestamp(), position);
```

Comment:
We need to log a position here since we don't want mixed records (some that have and some that don't have headers with consistency info) in the changelog

Reply:
Yep, I think that's correct anyhow, since deleting a record from a store is also an update to its state, and the position should uniquely correspond to whether you see the delete or not.